### PR TITLE
chore: Add privacy manifest to Auth0.Swift

### PR DESCRIPTION
**Pull request by poovamraj**
Privacy Manifest is required by Apple from May 1st. We have added this to our SDK in this PR

- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

### 📋 Changes

- Added PrivacyManifest file
- PrivacyManifest includes `NSPrivacyCollectedDataTypeUserID` with purpose as `NSPrivacyCollectedDataTypePurposeAppFunctionality`
- Added it to .podspec and Package.swift

### 📎 References

Screenshot 2024-04-12 at 13 10 20
https://developer.apple.com/app-store/app-privacy-details/#user-tracking
https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests#4144131

### 🎯 Testing

Repos with test integration
https://github.com/poovamraj/spm
https://github.com/poovamraj/carthage

### CODE REVIEW

File: Auth0.podspec [hunk 1]
```diff
@@ -41,6 +41,7 @@ Pod::Spec.new do |s|
   s.source           = { :git => 'https://github.com/auth0/Auth0.swift.git', :tag => s.version.to_s }
   s.social_media_url = 'https://twitter.com/auth0'
   s.source_files     = 'Auth0/*.swift'
+  s.resources        = "Sources/AuthFoundation/Resources/**/*"
```
Comment (Widcket) [inline Auth0.podspec:44]: The path `Sources/AuthFoundation/Resources/**/*` does not exist, and as such Cocoapods cannot find the file. Did you test the changes in a sample app that uses Cocoapods?

Screenshot 2024-04-12 at 13 55 01

File: Package.swift [hunk 2]
```diff
@@ -25,6 +25,7 @@ let package = Package(
             ],
             path: "Auth0",
             exclude: ["Info.plist"],
+            resources: [.copy(./Resources/PrivacyInfo.xcprivacy)]
```
Comment (Widcket) [inline Package.swift:28]: The syntax is invalid and does not compile. Did you test the changes on a sample app that uses the Swift Package Manager?

Screenshot 2024-04-12 at 13 54 37

File: Resources/PrivacyInfo.xcprivacy [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	<array/>
+	<key>NSPrivacyCollectedDataTypes</key>
+	<array>
+        <dict>
+			<key>NSPrivacyCollectedDataTypePurposes</key>
+			<array>
+				<string>NSPrivacyCollectedDataTypePurposeAppFunctionality</string>
+			</array>
+			<key>NSPrivacyCollectedDataType</key>
+			<string>NSPrivacyCollectedDataTypeUserID</string>
```
Comment (Widcket) [inline Resources/PrivacyInfo.xcprivacy:19]: This is missing `NSPrivacyCollectedDataTypeTracking` (`false`), and `NSPrivacyCollectedDataTypeLinked` (`true`).
Did you verify that the PDF privacy report gets generated with the correct content from an app archive?

**Review by Widcket — CHANGES_REQUESTED** [review]
This is missing the changes needed to support apps that use Carthage to install Auth0.swift.
Did you test the changes in a sample app that uses Carthage?

**Review by Widcket — APPROVED** [review]

**Comment by wongk** [comment]
This change is causing an error when building our application.
```
error: Multiple commands produce '/Users/expo/Library/Developer/Xcode/DerivedData/HealthCoach-cadkgkgehnezpjdtijikapiciihy/Build/Intermediates.noindex/ArchiveIntermediates/HealthCoach/InstallationBuildProductsLocation/Applications/HealthCoach.app/PrivacyInfo.xcprivacy'
```

Please see: https://stackoverflow.com/a/78344724. The podspec should use the following format:
```
"resource_bundles": {"LibName": "LibName/PrivacyInfo.xcprivacy"}
```

**Pull request merged**